src/windows/InstanceMutex.cpp

```diff
@@ -0,0 +1,32 @@
+#include "InstanceMutex.hpp"
+
+void InstanceMutex::Close()
+{
+	if (m_handle)
+	{
+		CloseHandle(m_handle);
+		m_handle = NULL;
+	}
+}
+
+HRESULT InstanceMutex::Init()
+{
+#ifdef DISABLE_SINGLE_INSTANCE
+	return 0;
+#else
+
+	SetLastError(NO_ERROR);
+	m_handle = CreateMutex(NULL, TRUE, L"DiscordMessenger");
+
+	const DWORD error = GetLastError();
+	if (error == ERROR_ALREADY_EXISTS)
```

Comment:
Check for the presence of `m_handle` instead of resetting the last error status.